### DISCUSSION TITLE
How can I locate binding errors in Avalonia?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#13011] [Q&A] [answered]
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/dae591b2-fc2a-457a-814a-e144870c58d0)
Recently I found that the output of visual studio reports errors about binding.

However, I don't know where the error actually locates on.

In spite of it seems my program runs well but I want to find the errors and clear them up.

How can I achieve this?

**Comment by maxkatz6** [comment, marked as answer]
These errors can be safely ignored, as it's a normal situation when control was just created and has DataContext null. See https://github.com/AvaloniaUI/Avalonia/issues/5762

**Reply by kyuranger** [reply]
All right, got it. Thank you so much.

**Comment by thevortexcloud** [comment]
For future reference, Avalonia often throws actual exceptions if the binding is a real problem. In which case Visual Studio will at least show the file the error is happening in if it configured to stop on the exception. But I agree, tracking it down without a debugger is not straightforward.

**Comment by timunie** [comment]
also compiled bindings help a lot. So if not enabled yet, I suggest to do it asap :-)

**Reply by kyuranger** [reply]
I didn't like compiled bindings in the past, so I always turned them off. However, recently I heard that compiled bindings can be helpful for program performance, and I will plan to turn it back on in the coming days.

**Reply by thevortexcloud** [reply]
There is no real reason to not use them if you can. They have several advantages. Especially if you plan to use AOT.
